modules/codegen/src/smithy4s/codegen/Renderer.scala

```diff
@@ -594,6 +592,27 @@ private[codegen] class Renderer(compilationUnit: CompilationUnit) { self =>
 
     def renderAlgParams =
       s"$renderInput, $renderError, $renderOutput, $renderStreamedInput, $renderStreamedOutput"
+
+    def renderAlgParamsWithImports(genName: String) = {
```

Comment:
the implementation of this should look like this : 

``` 
import op._ 
line"$input, $errors, $output, $streamedInput, $streamedOutput"
```

Reply:
problem I have is to return a Line , I must include the imports for Error , which require the Service Name . The problem is the Operation doesn't have access to the Service name, hence I was passing it in .
Should I leave out the Error imports and add them in via addImports function (which kind of returns us back to this ad hoc style..)

Reply:
The operations (and their error coproducts) are rendered within the service's companion object, so as long as we `import MyService._`  in the file of of the service, it should be okay really